src/garage/torch/algos/vpg.py

```diff
@@ -197,14 +194,26 @@ def _get_baselines(self, path):
         else:
             return torch.Tensor(self.baseline.predict(path))
 
-    def _filter_valids(self, tensor, valids):
-        return [tensor[i][:valids[i]] for i in range(len(valids))]
-
     def process_samples(self, itr, paths):
         """Process sample data based on the collected paths."""
         for path in paths:
             path['returns'] = special.discount_cumsum(path['rewards'],
                                                       self.discount)
+
+        valids, obs, actions, rewards = [], [], [], []
+        for path in paths:
+            valids.append(len(path['actions']))
+            obs.append(
```

Comment:
```python
obs = [_add_padding_2d(path['observations'], self.max_path_length) for path in paths]
```
...and similarly for returns, actions, rewards, etc.

Reply:
fixed